### PR TITLE
feat: semantic-health baseline watermark (v1.38.0)

### LINKED ISSUE #147: Change request: Support baselining for semantic health advisory warnings
Problem: The semantic-health pass (o2b brain doctor / o2b brain health) emits two advisory warnings - concept-gap and batch-concept-inflation - that are computed fresh on every run from the full vault history. When a vault is bulk-seeded with many preferences in a short window (as happens during initial import or migration), those warnings fire permanently: concept-gap has no time component at all, and batch-concept-inflation is anchored to the stored confirmed_at timestamps rather than to the current clock, so clock time does not age them off.

These warnings are benign (advisory only, verdict watch), but they are also immovable without deleting or editing real memory - there is no way to acknowledge a historical burst and stop it recurring in the report. To put it bluntly, we are stuck with them!

Requested enhancement: Add an optional "baseline" / "acknowledge-before" watermark (e.g. a health.silence_before date in _brain.yaml, or a small command that records one) such that any finding whose underlying entries are entirely older than the watermark is excluded from the report. The intent is to let the user declare "I've seen and accepted the state up to this date" so a one-time seed burst no longer keeps the verdict pinned at watch, while findings tied to entries after the watermark continue to surface normally. No change to how findings are detected or stored - only to which ones are surfaced.

### PR DESCRIPTION
**Pull request by solaitken**
## Summary

Operators can now acknowledge vault history up to a chosen date so that one-time bulk seeds stop pinning the semantic-health verdict at `watch` forever. A new `health.silence_before` watermark in `_brain.yaml` (managed by `o2b brain health-baseline set/get/clear`) filters advisory findings whose underlying entries are entirely older than the watermark out of the surfaced report, while detection, storage, and every finding tied to newer entries stay exactly as they are.

Closes #147.

## Why this matters

```mermaid
flowchart LR
    A[Bulk seed of preferences] --> B[concept-gap and batch-inflation fire]
    B --> C{Watermark set?}
    C -- no --> D[Verdict pinned at watch forever]
    C -- yes --> E[Findings entirely before baseline suppressed]
    E --> F[Verdict reflects current state]
    E --> G[Report prints suppressed count and baseline date]
    B -.entries after baseline.-> F
```

Before this change the only way to clear a permanently firing advisory was deleting or editing real memory. Now acknowledgement is a first-class, explicit, reversible operation that never touches stored data.

## Concrete benefits

- A one-time seed burst no longer keeps `o2b brain health` at `watch`; the verdict is computed from the surfaced findings only.
- Suppression is never silent: the report prints `suppressed: N finding(s) older than baseline <date>` and carries an additive `suppressed` object (mirrored snake_case on MCP `brain_health`).
- Conservative by construction: an entry with no parseable timestamp counts as newer than any watermark, so the filter can only hide what is provably old; a concept-gap term with even one post-baseline mention surfaces with its full frequency.
- Byte-identical when unset: no watermark means no new output, fields, or behavior anywhere (CLI text, CLI JSON, MCP), proven by regression tests.
- Invalid watermark values fail loudly at config load; the CLI setter validates input, exits 2 on usage errors, and emits `{ok:false}` envelopes on `--json` operational failures.
- The `_brain.yaml` upsert follows the repository convention (file lock plus atomic temp-file rename) and preserves unrelated config content byte-for-byte.
- Consolidation bonus: four duplicate detector-local ISO timestamp parsers collapsed into one shared `iso-time` module.

## What ships

- `src/core/brain/health/iso-time.ts` - shared ISO parsing helpers (single source, previously four copies).
- `src/core/brain/health-baseline.ts` - watermark read/write with a pure YAML upsert transform.
- `src/cli/brain/verbs/health-baseline.ts` - `o2b brain health-baseline set <date>|now / get / clear`.
- `src/core/brain/health/reconcile.ts` - surfacing-only baseline filter, additive `suppressed` report field, verdict over surfaced findings.
- `src/core/brain/policy.ts`, `types.ts` - `health.silence_before` config parsing and validation.
- `src/cli/brain/verbs/health.ts`, `src/mcp/brain/health-tools.ts` - explicit suppression line in text output and additive MCP mirror.
- Tests beside every change, including exact-boundary cases (equal-instant, straddling burst) and MCP presence/absence of `suppressed`.

## Test plan

- [ ] `bun run fmt` - no changes
- [ ] `bun run lint` - exactly 134 warnings, 0 errors (baseline held)
- [ ] `bun run typecheck` - clean
- [ ] `bun test` - 6803 pass, 0 fail
- [ ] `bun run scripts/sync-version.ts --check` - manifests in sync
- [ ] Temp-vault smoke: `set 2026-01-01` suppresses a pre-baseline burst and prints the suppressed line; `clear` restores the full report; unset vault output byte-identical to v1.37.0

## Summary by CodeRabbit

* **New Features**
  * Added semantic-health baselining via `o2b brain health-baseline set|get|clear` (watermark stored in `_brain.yaml`).
  * Health reports now suppress applicable advisory findings older than the configured watermark and include a breakdown of how many were suppressed plus the baseline date.
  * The `brain_health` MCP tool output now includes suppression details when baselining hides findings.

* **Documentation**
  * Updated README, CLI reference, MCP safety notes, and changelog for version 1.38.0.
  * Added help/usage text for the new `health-baseline` command.

* **Release**
  * Bumped package and plugin versions to 1.38.0.

### CODE REVIEW

File: src/core/brain/health-baseline.ts [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+  lines.splice(childIdx, 1);
+  // Drop a now-empty `health:` header (plus the blank lines it left behind)
+  // so clearing a CLI-created block returns the file to its prior shape.
+  if (findHealthChild(lines, headerIdx, /^[ \t]+\S/) < 0) {
+    let removeTo = headerIdx + 1;
+    while (removeTo < lines.length && lines[removeTo]!.trim() === "") removeTo++;
+    lines.splice(headerIdx, removeTo - headerIdx);
+  }
+  return lines.join("\n");
+}
```
[the remark this answers is not in the document]
Comment (solaitken) [reply src/core/brain/health-baseline.ts:129]: Both halves confirmed valid by failing tests first, fixed in d5b9cbca: the header regex now tolerates a trailing CR (`/^health:[ \t]*\r?$/`), all generated lines use the file's detected EOL style, a replaced `silence_before` line keeps its own indent and trailing CR, and a newly inserted line derives its indent from the block's first sibling child (two-space fallback only for an empty or new block). Untouched lines still round-trip byte-for-byte. Three new tests cover CRLF upsert-in-place, CRLF block creation, and a 4-space-indented block staying parseable; suite is 6806 pass, 0 fail.

File: src/core/brain/health-baseline.ts [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+ * (headerIdx), or `null` when the block has no children yet (empty or
+ * about to be populated for the first time). `parseBrainYaml` requires all
+ * siblings in a block to share one indent, so any inserted key must match
+ * whatever indent the block already established.
+ */
+function detectHealthSiblingIndent(lines: ReadonlyArray<string>, headerIdx: number): string | null {
+  for (let i = headerIdx + 1; i < lines.length; i++) {
+    const line = lines[i]!;
+    if (line.trim() === "") continue;
+    if (!/^[ \t]/.test(line)) break;
```
[the remark this answers is not in the document]
Comment (solaitken) [reply src/core/brain/health-baseline.ts:146]: Confirmed valid with failing tests first, fixed in 88f3c123: the sibling scan now skips lines whose trimmed form is empty or starts with #, matching parseBrainYaml's own comment handling, so an indented comment no longer donates its indent and an unindented comment no longer ends the scan early. Two new tests cover both failure modes; suite is 6808 pass, 0 fail.

**Review by itechmeat — APPROVED** [review]
